elastic/beats
# Add basic configuration for the APM tracer

## What does this PR do?

This adds configuration for the APM tracer, and sets it on the `Beat` object so that every Beat can use it.
Libbeat is responsible for correct initialization and closing.

A top level `instrumentation` config option is added to `libbeat.yml` / `reference.yml` to set it up, disabled by default.

## Why is it important?

Dogfooding!

## Checklist



## How to test this PR locally

Run any beat with configuration enabled and disabled, alongside with APM Server, Elasticsearch and Kibana. Observe `output` transactions (or lack thereof when disabled) in the Kibana APM tab, coming from the libbeat publisher.

## Related issues

Closes elastic/beats#18360 

An APM Server PR showing how to use the tracer from the Beat is coming up soon.
